Add spec v0.6.0 release blog (#129)
open-feature/openfeature.dev: blog/2023-05-24-spec-0-6-0.md

---
slug: 'spec-0.6.0'
title: 'Specification 0.6.0'
date: 2023-05-24
authors: [beeme1mr]
description: 'Announcing version 0.6.0 of the OpenFeature specification'

> keywords: [release, '0.6.0', openFeature, specification, spec]

Reviewer: ```suggestion
keywords: [release, '0.6.0', openFeature, specification, spec, events, shutdown, metadata]
```

> tags: [spec, specification, sdk]

Reviewer: ```suggestion
tags: [spec, specification, sdk, events, shutdown, metadata]
```

draft: false
---

> We are excited to announce the availability of version 0.6.0 of the OpenFeature spec. This release brings many significant additions that benefit both provider developers and end users, while also establishing a strong foundation for first-class client-side support in OpenFeature.

Reviewer: ```suggestion
We are excited to announce the availability of version 0.6.0 of the OpenFeature spec. This release brings many significant additions that benefit both provider developers and application authors, while also establishing a strong foundation for first-class client-side support in OpenFeature.
```

Reviewer: Do you mean actual end-users, or people writing applications with OpenFeature? I suspect the latter.

<!--truncate-->

## What's new?

### Multi-provider support

> OpenFeature now supports multiple providers in a single application, allowing teams to scope flag evaluation to a particular flag management system. This makes it easier for developers to mix and match providers based on their needs, all from a unified SDK.

Reviewer: Really like this simple paragraph.

> To learn more, please refer to the [OpenFeature Enhancement Proposal](https://github.com/open-feature/ofep/blob/main/OFEP-provider-client-mapping.md) or [spec change](https://openfeature.dev/specification/sections/flag-evaluation/#requirement-113).

### Flag metadata

> A feature flag management system typically has a wealth of information about a particular feature flag, such as a configuration URL or the originating Jira issue. Wouldn't it be useful if you were able to access it within the OpenFeature SDK? With flag metadata, providers can now annotate feature flags with additional data.

Reviewer: Totally optional.

> A feature flag management system typically has a wealth of information about a particular feature flag, such as a configuration URL or the originating Jira issue. Wouldn't it be useful if you were able to access it within the OpenFeature SDK? With flag metadata, providers can now annotate feature flags with additional data.

Author: I see where you're going with this but I don't think it's necessary.

> For further information, please consult the [OpenFeature Enhancement Proposal](https://github.com/open-feature/ofep/pull/46) or [spec change](https://openfeature.dev/specification/sections/providers#24-initialization).

### Events

Developers can now incorporate custom logic that responds to important events, such as changes in flag rulesets. This allows for the reevaluation of feature flags without user interaction. These events can be leveraged to instantly update portions of a user interface or dynamically adjust the log level on a server. Event types include flag ruleset changes, provider readiness, and error events.

> For more details, please refer to the [OpenFeature Enhancement Proposal](https://github.com/open-feature/ofep/blob/main/007-OFEP-flag-change-events.md) or [spec change](https://openfeature.dev/specification/sections/events).

## Conclusion

We can't wait to see what you do with these new features. SDKs have started implementing the updated spec and should be available shortly. To get check the progress or get involved, checkout [these issues](https://github.com/search?q=org%3Aopen-feature+label%3Av0.6.0&type=issues&s=created&o=asc) on GitHub.

> And, as teased in the intro, the team is already hard at work defining the spec for [client-side feature flags](https://github.com/open-feature/spec/pull/171).

Reviewer: ```suggestion
And, as teased in the intro, the team is already hard at work enhancing the spec to better support [client-side feature flags](https://github.com/open-feature/spec/pull/171).
```

Reviewer: Also optional.
